src/v0/destinations/dynamic_yield/util.js

```diff
@@ -0,0 +1,64 @@
+const { constructPayload } = require('../../util');
+const { ConfigCategory, mappingConfig } = require('./config');
+
+const ecomEventsMapping = {
+  product_added: 'Add to Cart',
+  product_removed: 'Remove from Cart',
+  product_added_to_wishlist: 'Add to Wishlist',
+  order_completed: 'Purchase',
+};
+
+const populateProperties = (messageProperties, event) => {
+  const properties = {
+    dyType: `${ecomEventsMapping[event].toLowerCase().trim().replace(/\s+/g, '-')}-v1`,
+    value: Number(parseFloat(messageProperties.price).toFixed(2)),
+    currency: messageProperties.currency,
+  };
+  if (event === 'order_completed') {
+    properties.uniqeTransactionId = messageProperties.order_id;
+    properties.value =
+      messageProperties.value || messageProperties.revenue || messageProperties.price;
+  } else {
+    properties.quantity = messageProperties.quantity;
+    properties.productId = messageProperties.sku || messageProperties.product_id;
+  }
+  if (messageProperties.products && Array.isArray(messageProperties.products)) {
+    properties.cart = [];
+    const { products } = messageProperties;
+    products.forEach((product) => {
+      const cartObj = {
+        productId: product.sku || product.product_id,
+        quantity: product.quantity,
+        itemPrice: product.price,
+      };
+      properties.cart.push(cartObj);
+    });
+  }
+  return properties;
+};
+
+const preparePayload = (message, event) => {
+  const payload = constructPayload(message, mappingConfig[ConfigCategory.TRACK.name]);
```

Comment:
Move `constructPayload` to `trackResponseBuilder` & clean the corresponding import of config.js

Reply:
Done